internal/manifests/collector/deployment.go

```diff
@@ -19,54 +19,66 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"github.com/open-telemetry/opentelemetry-operator/internal/api/convert"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests"
 	"github.com/open-telemetry/opentelemetry-operator/internal/manifests/manifestutils"
 	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
 )
 
 // Deployment builds the deployment for the given instance.
-func Deployment(params manifests.Params) *appsv1.Deployment {
-	name := naming.Collector(params.OtelCol.Name)
-	labels := manifestutils.Labels(params.OtelCol.ObjectMeta, name, params.OtelCol.Spec.Image, ComponentOpenTelemetryCollector, params.Config.LabelsFilter())
+func Deployment(params manifests.Params) (*appsv1.Deployment, error) {
```

Comment:
I think after this change and #2516 go in, i will modify these signatures to not take in the whole `manifests.Params` object, but only the object they need (i.e. collector in this case)

Reply:
- https://github.com/open-telemetry/opentelemetry-operator/issues/2592